### PR TITLE
Destroy gpu device on root.destroy

### LINKED ISSUE #616: Destroy a device when destroying a root, but only when the root was responsible for its creation.
```ts
const device = ...;
const root = tgpu.initFromDevice({ device });

root.destroy(); // Device is still alive
```

```ts
const root = await tgpu.init();

root.destroy(); // Device is destroyed along with the root
```

### PR DESCRIPTION
**Pull request by mhawryluk**
closes #616

### CODE REVIEW

**Review by reczkok — APPROVED** [review]
LGTM 🛠️👷‍♂️

File: apps/typegpu-docs/src/content/docs/fundamentals/roots.mdx [hunk 1]
```diff
@@ -59,7 +59,8 @@ untyped value of a typed resource, use the `root.unwrap` function.
 
 ## Destroying resources
 
-Calling `root.destroy()` will destroy all resources created with it.
+Calling `root.destroy()` will destroy all resources created with it. 
+It will also destroy the underlying WebGPU device, if it wasn't originally passed in via the `initFromDevice` function.
```
Comment (iwoplaza) [inline apps/typegpu-docs/src/content/docs/fundamentals/roots.mdx]: Only true once we release 0.3
We could either postpone the merging of this PR until then, or only add this line of documentation to the 0.3 branch.
```suggestion
```

**Review by iwoplaza — APPROVED** [review]
Great job! Let's not merge it until we can decide what to do with the docs.

File: apps/typegpu-docs/src/content/docs/fundamentals/roots.mdx (hunk 1 above)
Comment (mhawryluk) [reply apps/typegpu-docs/src/content/docs/fundamentals/roots.mdx]: i choose number 2

File: apps/typegpu-docs/src/content/docs/fundamentals/roots.mdx [hunk 2]
```diff
@@ -59,7 +59,7 @@ untyped value of a typed resource, use the `root.unwrap` function.
 
 ## Destroying resources
 
-Calling `root.destroy()` will destroy all resources created with it.
+Calling `root.destroy()` will destroy all resources created with it.
```
Comment (mhawryluk) [inline apps/typegpu-docs/src/content/docs/fundamentals/roots.mdx]:
```suggestion
Calling `root.destroy()` will destroy all resources created with it.
```

**Pull request merged**